src/lib/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,145 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+
+import "./Pagination.scss";
+import { Button, Icon, Input } from "@canonical/react-components";
+
+export interface PaginationProps {
+  currentPage: number | undefined;
+  error?: string;
+  isLoading: boolean;
+  onInputBlur: () => void;
+  onInputChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  onNextClick: () => void;
+  onPreviousClick: () => void;
+  pageNumber: number | undefined;
+  totalPages: number;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  currentPage,
+  error,
+  isLoading,
+  onInputBlur,
+  onInputChange,
+  onNextClick,
+  onPreviousClick,
+  pageNumber,
+  totalPages,
+}: PaginationProps) => {
+  return (
+    <nav aria-label="pagination" className="p-pagination">
+      <span className="p-pagination--items">
+        <Button
+          aria-label="Previous page"
+          className="p-pagination__link--previous"
+          disabled={currentPage === 1 || isLoading}
+          onClick={onPreviousClick}
+          type="button"
+        >
+          <Icon name="chevron-down" />
+        </Button>
+        <strong>Page </strong>{" "}
+        <Input
+          aria-label="page number"
+          className="p-pagination__input"
+          disabled={isLoading}
+          error={error}
+          min={1}
+          onBlur={onInputBlur}
+          onChange={onInputChange}
+          required
+          type="number"
+          value={pageNumber}
+        />{" "}
+        <strong className="u-no-wrap"> of {totalPages}</strong>
+        <Button
+          aria-label="Next page"
+          className="p-pagination__link--next"
+          disabled={currentPage === totalPages || isLoading}
+          onClick={onNextClick}
+          type="button"
+        >
+          <Icon name="chevron-up" />
+        </Button>
+      </span>
+    </nav>
+  );
+};
+
+export interface PaginationContainerProps {
+  currentPage: PaginationProps["currentPage"];
+  debounceInterval: number;
```

Comment:
Debouncing should be handled outside of this component IMO. Ideal UX is an immediate change in the UI, debouncing should be handled by he parent component or a custom hook.

Besides, it's a good practice to separate the logic associated with API calls from UI components, it will help with reusability.

```suggestion
```